pennylane/noise/conditionals.py

```diff
@@ -0,0 +1,386 @@
+
+    Returns:
+        :class:`NoiseConditional <pennylane.noise.NoiseConditional>`: a boolean function
+        represented as ``WiresEq``, which evaluates to ``True`` if a given wire is equal to
+        specified set of wires.
+
+    Raises:
+        ValueError: if the wire set cannot be computed from ``wires``.
+
+    **Example**
+
+    One may use ``wires_eq`` with a given sequence of wires which are used as a wire set:
+
+    >>> cond_func = qml.noise.wires_eq(0)
+    >>> cond_func(0)
+    True
+    >>> cond_func(qml.RY(1.23, wires=[3]))
+    False
+
+    Additionally, if an :class:`Operation <pennylane.operation.Operation>` is provided,
+    its ``wires`` are extracted and used to build the wire set:
+
+    >>> cond_func = qml.noise.wires_in(qml.RX(1.0, "dino"))
+    >>> cond_func(qml.RZ(1.23, wires="dino"))
+    True
+    >>> cond_func("eve")
+    False
+    """
+    return NoiseConditional(
+        lambda x: _get_wires(x).issubset(_get_wires(wires)), f"WiresEq({wires})"
+    )
+
+
+def _get_ops(val):
+    """Computes the class for a given argument from its string name, instance,
+    or a sequence of them.
+
+    Args:
+        val (Union[str, Operation, Iterable]): object to be used
+            for building the wire set.
+
+    Returns:
+        tuple[class]: tuple of :class:`Operation <pennylane.operation.Operation>`
+        classes corresponding to val.
+    """
+    vals = val if isinstance(val, (list, tuple, set, qml.wires.Wires)) else [val]
+    return tuple(
+        (
+            getattr(qml.ops, val)
+            if isinstance(val, str)
+            else (val if isclass(val) else getattr(val, "__class__"))
+        )
+        for val in vals
+    )
+
+
+def op_eq(ops):
+    """Builds a ``Conditional`` as a boolean function for evaluating
+    if a given operation is equal to the specified operation.
+
+    Args:
+        ops (str, Operation, Union(list[str, Operation])): string
+            representation or instance of the operation.
+
+    Returns:
+        :class:`NoiseConditional <pennylane.noise.NoiseConditional>`: a boolean function
+        represented as ``OpEq``, which evaluates to ``True`` if a given operation is
+        equal to the specified set of operation(s), irrespective of wires they act on.
+
+    **Example**
+
+    One may use ``op_eq`` with a string representation of the name of the operation:
+
+    >>> cond_func = qml.noise.op_eq("RX")
+    >>> cond_func(qml.RX(1.23, wires=[0]))
+    True
+    >>> cond_func(qml.RZ(1.23, wires=[3]))
+    False
+    >>> cond_func("CNOT")
+    False
+
+    Additionally, an instance of :class:`Operation <pennylane.operation.Operation>`
+    can also be provided:
+
+    >>> cond_func = qml.noise.op_eq(qml.RX(1.0, "dino"))
+    >>> cond_func(qml.RX(1.23, wires=["eve"]))
+    True
+    >>> cond_func(qml.RY(1.23, wires=["dino"]))
+    False
+    """
+    op_cls = _get_ops(ops)
+    op_repr = str([getattr(op, "__name__") for op in op_cls])[1:-1]
```

Comment:
I would expect this to be false:

```python
cond_func = qml.noise.op_eq(qml.RX(1,1)@qml.RY(1,0))
print(cond_func(qml.RZ(1, 1)@qml.RZ(1,0)))

>>> True

Reply:
Ahh! This was not supposed to support arithmetic operators, but I have tried to add that support now. So this should work.